### ISSUE TITLE
`Style/FormatString` cop breaks code

### ISSUE THREAD

**Issue by michiomochi** [rubocop-hq/rubocop#7130]
Rubocop auto-corrector breaks code by `Style/FormatString` cop.

Before code is

```
def hoge
  "%08x-%04x-%04x-%04x-%04x%08x" % ary
end
```

Execute auto-correct

```
> bundle exec rubocop -a
```

After code is

```
def hoge
  format("%08x-%04x-%04x-%04x-%04x%08x", ary)
end
```

## Expected behavior

No effort.

## Actual behavior

Fix to unexpected code.

## RuboCop version

```
> bundle exec rubocop -v
0.71.0
```

**Comment by pocke** [comment]
`String#%` can receive an array, but `format` cannot receive it.

```ruby
"%02x" % 1 # => "01"
"%02x" % [1] # => "01"
"%02x-%02x" % [1, 2] # => "01-02"

format("%02x", 1)# => "01"
format("%02x", [1])# => TypeError
format("%02x-%02x", [1, 2]) # => TypeError
format("%02x-%02x", 1, 2) # => "01-02"
```


I think we should mark the auto-correction as unsafe.

**Comment by tejasbubane** [comment]
@pocke Is is a good idea to not register offense when second argument to `String#%` is a variable?

**Comment by pocke** [comment]
> @pocke Is is a good idea to not register offense when second argument to String#% is a variable?

It makes sense, but I prefer changing auto-correct safety.
Because if the cop does not register offense, user has no way to know `format` method.


---


By the way, I got other ideas.

First, ignore only auto-correction when the argument is a variable.
The cop registers an offesne for `String%` with a variable but does not apply auto-correction.



Second, I guess we can use splat operator for a variable, like `format("str", *variable)`.
I think it is a better way, but I'm not sure splat operator does not change meaning anytime.

**Comment by tejasbubane** [comment]
> I guess we can use splat operator for a variable, like `format("str", *variable)`

This will be an issue if variable is a hash. First approach looks better.
